From 1ae1a33af61e8177ea85702bb378131b244f99a2 Mon Sep 17 00:00:00 2001
From: Dave Gerlach <d-gerlach@ti.com>
Date: Mon, 25 Aug 2014 18:18:39 -0500
Subject: [PATCH 1101/1587] ARM: OMAP2+: opp33xx: Change OPP50 initialization

Modify how we initialize 300MHz OPP, make sure we do not accidentally enable
one and then immediately disable it if the other 300MHz OPP is not
selected by eFuse.

Signed-off-by: Dave Gerlach <d-gerlach@ti.com>
[zou: Original patch taken from
ti-sdk-am335x-evm-08.00.00.00-Linux-x86-Install.bin]
Signed-off-by: Cao Zou <cao.zou@windriver.com>
---
 arch/arm/mach-omap2/opp33xx_data.c |   21 +++++++++++----------
 1 files changed, 11 insertions(+), 10 deletions(-)

diff --git a/arch/arm/mach-omap2/opp33xx_data.c b/arch/arm/mach-omap2/opp33xx_data.c
index 1de589e..0e07186 100644
--- a/arch/arm/mach-omap2/opp33xx_data.c
+++ b/arch/arm/mach-omap2/opp33xx_data.c
@@ -123,20 +123,21 @@ int __init am33xx_opp_init(void)
 			opp_def_list_update_opp_voltage(am33xx_es2_x_opp_list,
 				ARRAY_SIZE(am33xx_es2_x_opp_list),
 				"mpu", 300000000, 1100000);
+			opp_def_list_enable_opp(am33xx_es2_x_opp_list,
+				ARRAY_SIZE(am33xx_es2_x_opp_list),
+				"mpu", 300000000, true);
+		} else if (max_freq & EFUSE_OPP_50_300MHZ_BIT) {
+			opp_def_list_enable_opp(am33xx_es2_x_opp_list,
+				ARRAY_SIZE(am33xx_es2_x_opp_list),
+				"mpu", 300000000, true);
+		} else {
+			opp_def_list_enable_opp(am33xx_es2_x_opp_list,
+				ARRAY_SIZE(am33xx_es2_x_opp_list),
+				"mpu", 300000000, false);
 		}
 
 		opp_def_list_enable_opp(am33xx_es2_x_opp_list,
 			ARRAY_SIZE(am33xx_es2_x_opp_list),
-			"mpu", 300000000,
-			(max_freq & EFUSE_OPP_50_300MHZ_BIT) ? true : false);
-
-		opp_def_list_enable_opp(am33xx_es2_x_opp_list,
-			ARRAY_SIZE(am33xx_es2_x_opp_list),
-			"mpu", 300000000,
-			(max_freq & EFUSE_OPP_100_300MHZ_BIT) ? true : false);
-
-		opp_def_list_enable_opp(am33xx_es2_x_opp_list,
-			ARRAY_SIZE(am33xx_es2_x_opp_list),
 			"mpu", 600000000,
 			(max_freq & EFUSE_OPP_100_600MHZ_BIT) ? true : false);
 
-- 
1.7.5.4

